include/vicon/udp.h

```diff
@@ -399,13 +385,13 @@ class UDPClient
          */
         using namespace units::length;
         using namespace units::angle;
-        m_ObjectData[id].update(frameNumber,
-                                { { millimeter_t(data.position[0]),
-                                    millimeter_t(data.position[1]),
-                                    millimeter_t(data.position[2])},
-                                  { radian_t(data.attitude[2]),
-                                    radian_t(data.attitude[0]),
-                                    radian_t(data.attitude[1]) } });
+        m_ObjectData.at(objectName).update(frameNumber,
```

Comment:
you could save a second lookup by not having ``pos`` be const, setting it to ``m_ObjectData.emplace(...).first`` and calling ``pos->second.update(...)``

Reply:
Good idea